state/src/replication/state_machine/snapshot.rs

```diff
@@ -75,13 +75,17 @@ impl SnapshotInfo {
 
 impl RaftSnapshotBuilder<TypeConfig> for StateMachine {
     async fn build_snapshot(&mut self) -> Result<Snapshot<TypeConfig>, RaftStorageError<NodeId>> {
+        // Create the lock file
+        self.create_snapshot_lock().await.map_err(new_snapshot_error)?;
```

Comment:
if the snapshot directory doesn't exist, this method will fail - we should have it create the snapshot directory if it doesn't exist like `take_db_snapshot`

Reply:
done, added it to `create_snapshot_lock`